src/api/cow/index.ts

```diff
@@ -329,48 +347,108 @@ export class CowApi {
     }
   }
 
-  private async getProfileApiBaseUrl(): Promise<string> {
-    const chainId = await this.context.chainId
-    const baseUrl = this.PROFILE_API_BASE_URL[chainId]
-
-    if (!baseUrl) {
-      throw new CowError(`Unsupported Network. The ${this.API_NAME} API is not deployed in the Network ` + chainId)
-    } else {
-      return baseUrl + '/v1'
-    }
-  }
-
-  private async fetch(url: string, method: 'GET' | 'POST' | 'DELETE', data?: unknown): Promise<Response> {
-    const baseUrl = await this.getApiBaseUrl()
+  private async fetch(
+    url: string,
+    method: 'GET' | 'POST' | 'DELETE',
+    baseUrl: string,
+    data?: unknown
+  ): Promise<Response> {
     return fetch(baseUrl + url, {
       headers: this.DEFAULT_HEADERS,
       method,
       body: data !== undefined ? JSON.stringify(data) : data,
     })
   }
 
-  private async fetchProfile(url: string, method: 'GET' | 'POST' | 'DELETE', data?: unknown): Promise<Response> {
-    const baseUrl = await this.getProfileApiBaseUrl()
+  private async fetchProfile(
+    url: string,
+    method: 'GET' | 'POST' | 'DELETE',
+    baseUrl: string,
+    data?: unknown
+  ): Promise<Response> {
     return fetch(baseUrl + url, {
       headers: this.DEFAULT_HEADERS,
       method,
       body: data !== undefined ? JSON.stringify(data) : data,
     })
   }
 
-  private post(url: string, data: unknown): Promise<Response> {
-    return this.fetch(url, 'POST', data)
+  private async post(url: string, data: unknown, options: Options = {}): Promise<Response> {
+    const { chainId: networkId, isDevEnvironment } = options
+    const prodUri = getGnosisProtocolUrl(false)
+    const barnUri = getGnosisProtocolUrl(true)
+    const chainId = networkId || (await this.context.chainId)
+    let response
+    if (isDevEnvironment === undefined) {
+      try {
+        response = await this.fetch(url, 'POST', `${prodUri[chainId]}/v1`, data)
+      } catch (error) {
+        response = await this.fetch(url, 'POST', `${barnUri[chainId]}/v1`, data)
+      }
+      return response
+    } else {
+      const uri = isDevEnvironment ? barnUri : prodUri
+      return await this.fetch(url, 'POST', `${uri[chainId]}/v1`, data)
+    }
   }
 
-  private get(url: string): Promise<Response> {
-    return this.fetch(url, 'GET')
+  private async get(url: string, options: Options = {}): Promise<Response> {
```

Comment:
The `post` and `get` methods look very similar.
Would it be possible to refactor it?

Reply:
In fact, include in that `delete` and even the `getProfile` methods